ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/InMemoryKVParticipantStateIT.scala

```diff
@@ -85,7 +85,59 @@ class InMemoryKVParticipantStateIT extends AsyncWordSpec with AkkaBeforeAndAfter
       }
     }
 
-    "duplicate package removed from update after uploadPackages" in {
+    "provide two updates after uploadPackages with two archives" in {
+      val ps = new InMemoryKVParticipantState(participantId)
+      val rt = ps.getNewRecordTime()
+
+      val sourceDescription = Some("provided by test")
+      val archives = List(
+        DamlLf.Archive.newBuilder
+          .setHash("asdf")
+          .setPayload(ByteString.copyFromUtf8("AAAAAAAHHHHHH"))
+          .build,
+        DamlLf.Archive.newBuilder
+          .setHash("zxcv")
+          .setPayload(ByteString.copyFromUtf8("ZZZZZZZZZZZZZ"))
+          .build
+      )
+
+      for {
+        result <- ps
+          .uploadPackages(archives, sourceDescription)
+          .toScala
+        updateTuples <- ps.stateUpdates(beginAfter = None).take(2).runWith(Sink.seq)
+      } yield {
+        ps.close()
+        result match {
+          case UploadPackagesResult.Ok =>
+            succeed
+          case _ =>
+            fail("unexpected response to party allocation")
+        }
+        updateTuples.head match {
```

Comment:
Minor: this block and the following can probably be defined as a function with the offset as a parameter.

Reply:
Done